### PR TITLE
Documentation on Message Model BE, MessageThread Model BE, Thread.jsx FE and ThreadPage.jsx FE

### PR DESCRIPTION
**Pull request by estherchee**
- Document Message Model BE
- Document MessageThread Model BE
- Document Thread.jsx FE
- Document ThreadPage.jsx FE

Technical Requirement:
- Documentation to prepare java.doc

### CODE REVIEW

File: src/main/java/se/kth/sda/skeleton/chat/messages/MessageController.java [hunk 1]
```diff
@@ -7,9 +7,12 @@
 import se.kth.sda.skeleton.auth.AuthService;
 import se.kth.sda.skeleton.chat.threads.MessageThread;
 import se.kth.sda.skeleton.chat.threads.MessageThreadService;
-
 import java.util.List;
 
+/**
+ * Represents the controller layer (or the API). This exposes application functionality of Post as RESTful webservices.
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/chat/messages/MessageController.java]: Please replace `Post` with `MessageController`

File: src/main/java/se/kth/sda/skeleton/chat/messages/MessageService.java [hunk 2]
```diff
@@ -4,34 +4,52 @@
...
 import java.util.List;
-import java.util.Optional;
 
+/**
+ * Represents the service layer. It contains defined functionalities according to business logic for Messages.
+ */
 @Service
-public class MessageService {
+public class
+
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/chat/messages/MessageService.java:13]: remove extra line and put the class header on the same line

**Review by NeuralAlchemist — APPROVED** [review]
Approved, please fix minor changes mentioned by en-chi

File: src/main/java/se/kth/sda/skeleton/chat/threads/MessageThread.java [hunk 3]
```diff
@@ -6,8 +6,10 @@
 import javax.persistence.*;
 import java.util.List;
 
-/* To get around the issue of a thread being deleted, and both parties losing the messages,
-   we could explore creating a thread for each user with references to the same messages  */
+/**
+ * Represents a messageThread in a cuddly app as a JPA entity. This implementation of message will autogenerate a primary key of type
+ * {@link Long} to indicate the current Message entity. A Message has a {@link String} p1_email and p2_email.
+ */
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/chat/threads/MessageThread.java]: remove extra line

**Review by eeels22 — APPROVED** [review]
thnaks Pei!

**Review by deciocandido — COMMENTED** [review]
Looks really good

**Pull request merged**